code/game/objects/items/weapons/grenades/gravitygrenade.dm

```diff
@@ -0,0 +1,58 @@
+/obj/item/grenade/gravitygrenade
+	name = "Gravaton Grenade"
+	desc = "This grenade emmits a blast of negative gravatons, inverting gravity in the nearbye area for 30 seconds."
+	icon_state = "gravity"
+	item_state = "flashtool"
+	origin_tech = "magnets=4;combat=4"
+	///used if it is a clusterbuster, so it deletes if it fails to activate, and doesnt leave 10+ unactivated nades
+	var/clustersegment = FALSE
+	///How long untill gravity returns (In deciseconds)?
+	var/gravity_timer = 300
+
+/obj/item/grenade/gravitygrenade/prime()
+	update_mob()
+	var/area/area = get_area(src)
+	if(area.outdoors)
+		visible_message("<span class='warning'> The [src] will not work in such a big area!</span>")
+		unprime()
+		return
+	if(!is_teleport_allowed(area.z))
+		visible_message("<span class='warning'> The [src] seems to not work here!</span>")
+		unprime()
+		return
+	if(istype(area, /area/shuttle) || istype(area, /area/ruin))
+		visible_message("<span class='warning'> The [src] seems to not work here!</span>")
+		unprime()
+		return
+	if(area.has_gravity)
+		area.gravitychange(0, area)
+	else
+		area.gravitychange(1, area)
+
+	if(area.special_gravity)
+		area.special_gravity = FALSE
+	else
+		area.special_gravity = TRUE
+	playsound(loc, 'sound/weapons/wave.ogg', 60, 1)
+	addtimer(CALLBACK(null, .proc/unskrek_gravity, area), gravity_timer)
+	qdel(src)
+
+/obj/item/grenade/gravitygrenade/proc/unskrek_gravity(area/area)
+	if(area.has_gravity)
+		area.gravitychange(0, area)
+	else
+		area.gravitychange(1, area)
+
+	if(area.special_gravity)
+		area.special_gravity = FALSE
+	else
+		area.special_gravity = TRUE
```

Comment:
I recommend changing this to something like:
```dm
/obj/item/grenade/gravitygrenade/proc/invert_gravity(area/area)
	area.gravitychange(!area.has_gravity, area)
	area.special_gravity = !area.special_gravity
```
Takes advantage of the fact you're dealing with booleans. You don't have to change the logic inside the proc if you don't want, but the point is, you can use this to replace the duplicate code from line 27 to 35.

Reply:
Done!